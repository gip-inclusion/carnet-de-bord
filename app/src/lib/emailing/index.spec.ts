/**
 * @jest-environment jsdom
 */

import { createMail } from './';
import { render } from '@testing-library/svelte';

const url = {
	appUrl: 'some-url.com',
	accessKey: 'access-key',
	redirectUrl: 'redirect-to-url.com',
};

describe('creating email', () => {
	describe('login email', () => {
		it('render email template as html string', async () => {
			const email = await createMail({
				template: 'LoginRequest',
				params: [{ url, pro: { firstname: 'Pr√©nom', lastname: 'Nom' } }],
				render,
			});

			expect(email).toMatchSnapshot();
		});
	});
});
