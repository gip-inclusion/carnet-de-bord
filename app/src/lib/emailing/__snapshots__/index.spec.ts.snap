// Vitest Snapshot v1

exports[`creating email > accountCreatedByAdmin email > renders email template as html string 1`] = `
<div>
  <div>
    <p>
      Bonjour 
      Prénom Nom
      ,
    </p>
    <!--&lt;Greeting&gt;-->
     
    <p>
      Un compte vous a été créé sur l'application Carnet de bord.
    </p>
     
    <p>
      Votre identifiant de connexion est 
      <b>
        prenom.nom
      </b>
    </p>
     
    <p>
      Pour accéder à Carnet de bord, veuillez cliquer sur le lien ci-dessous :
    </p>
     
    <p
      style="padding-left: 20%;"
    >
      <a
        href="some-url.com/auth/jwt/access-key?url=redirect-to-url.com"
        rel="nofollow"
        style="background-color: rgb(106, 32, 174); font-size: 14px; font-family: Helvetica, Arial, sans-serif; font-weight: bold; text-decoration: none; padding: 8px 10px; color: rgb(255, 255, 255); border-radius: 5px; display: inline-block;"
      >
        <span>
          Accédez à Carnet de bord
        </span>
      </a>
    </p>
    <!--&lt;AccessButton&gt;-->
     
    <p>
      L'équipe Carnet de bord
    </p>
    <!--&lt;Footer&gt;-->
  </div>
  <!--&lt;AccountCreatedByAdmin&gt;-->
</div>
`;

exports[`creating email > accountRequest email > renders email template as html string 1`] = `
<div>
  <div>
    <p>
      Bonjour,
    </p>
    <!--&lt;Greeting&gt;-->
     
    <p>
      Une demande de création d'un compte pour 
      Prénom Nom
       a été reçue.
    </p>
     
    <b>
      Informations personnelles :
    </b>
     
    <ul>
      <li>
        Structure : 
        Meilleure structure
      </li>
       
      <li>
        Prénom : 
        Prénom
      </li>
       
      <li>
        Nom : 
        Nom
      </li>
       
      <li>
        Courriel : 
        prenom@nom.org
      </li>
       
      <li>
        téléphone : 
        06123456789
      </li>
       
      <li>
        position : 
        Président de l‘univers
      </li>
       
      <p>
        Cette demande a été soumise par 
        Prénom du demandeur Nom du demandeur
        .
      </p>
       
      <p>
        Vous recevez ce message parce que vous êtes manager de ce déploiement. Veuillez vous connecter
			à Carnet de bord pour confirmer cette inscription.
      </p>
    </ul>
     
    <p>
      Pour accéder à Carnet de bord, veuillez cliquer sur le lien ci-dessous :
    </p>
     
    <p
      style="padding-left: 20%;"
    >
      <a
        href="some-url.com/auth/jwt/access-key?url=redirect-to-url.com"
        rel="nofollow"
        style="background-color: rgb(106, 32, 174); font-size: 14px; font-family: Helvetica, Arial, sans-serif; font-weight: bold; text-decoration: none; padding: 8px 10px; color: rgb(255, 255, 255); border-radius: 5px; display: inline-block;"
      >
        <span>
          Accédez à Carnet de bord
        </span>
      </a>
    </p>
    <!--&lt;AccessButton&gt;-->
     
    <p>
      L'équipe Carnet de bord
    </p>
    <!--&lt;Footer&gt;-->
  </div>
  <!--&lt;AccountRequest&gt;-->
</div>
`;

exports[`creating email > accountRequestValidate email > renders email template as html string 1`] = `
<div>
  <div>
    <p>
      Bonjour 
      Prénom Nom
      ,
    </p>
    <!--&lt;Greeting&gt;-->
     
    <p>
      Votre demande d'inscription à Carnet de bord a été acceptée.
    </p>
     
    <p>
      Pour accéder à Carnet de bord, veuillez cliquer sur le lien ci-dessous :
    </p>
     
    <p
      style="padding-left: 20%;"
    >
      <a
        href="some-url.com/auth/jwt/access-key?url=redirect-to-url.com"
        rel="nofollow"
        style="background-color: rgb(106, 32, 174); font-size: 14px; font-family: Helvetica, Arial, sans-serif; font-weight: bold; text-decoration: none; padding: 8px 10px; color: rgb(255, 255, 255); border-radius: 5px; display: inline-block;"
      >
        <span>
          Accédez à Carnet de bord
        </span>
      </a>
    </p>
    <!--&lt;AccessButton&gt;-->
     
    <p>
      L'équipe Carnet de bord
    </p>
    <!--&lt;Footer&gt;-->
  </div>
  <!--&lt;AccountRequestValidate&gt;-->
</div>
`;

exports[`creating email > forgotLogin email > renders email template as html string 1`] = `
<div>
  <div>
    <p>
      Bonjour 
      Prénom Nom
      ,
    </p>
    <!--&lt;Greeting&gt;-->
     
    <p>
      Votre identifiant de connexion est 
      <b>
        prenom.nom
      </b>
    </p>
     
    <p>
      Pour accéder à Carnet de bord, veuillez cliquer sur le lien ci-dessous :
    </p>
     
    <p
      style="padding-left: 20%;"
    >
      <a
        href="some-url.com/auth/jwt/access-key?url=redirect-to-url.com"
        rel="nofollow"
        style="background-color: rgb(106, 32, 174); font-size: 14px; font-family: Helvetica, Arial, sans-serif; font-weight: bold; text-decoration: none; padding: 8px 10px; color: rgb(255, 255, 255); border-radius: 5px; display: inline-block;"
      >
        <span>
          Accédez à Carnet de bord
        </span>
      </a>
    </p>
    <!--&lt;AccessButton&gt;-->
     
    <p>
      L'équipe Carnet de bord
    </p>
    <!--&lt;Footer&gt;-->
  </div>
  <!--&lt;ForgotLoginRequest&gt;-->
</div>
`;

exports[`creating email > login email > render email template as html string 1`] = `
<div>
  <div>
    <p>
      Bonjour 
      Prénom Nom
      ,
    </p>
    <!--&lt;Greeting&gt;-->
     
    <p>
      Pour accéder à Carnet de bord, veuillez cliquer sur le lien ci-dessous :
    </p>
     
    <p
      style="padding-left: 20%;"
    >
      <a
        href="some-url.com/auth/jwt/access-key?url=redirect-to-url.com"
        rel="nofollow"
        style="background-color: rgb(106, 32, 174); font-size: 14px; font-family: Helvetica, Arial, sans-serif; font-weight: bold; text-decoration: none; padding: 8px 10px; color: rgb(255, 255, 255); border-radius: 5px; display: inline-block;"
      >
        <span>
          Accédez à Carnet de bord
        </span>
      </a>
    </p>
    <!--&lt;AccessButton&gt;-->
     
    <p>
      L'équipe Carnet de bord
    </p>
    <!--&lt;Footer&gt;-->
  </div>
  <!--&lt;LoginRequest&gt;-->
</div>
`;

exports[`creating email > managerOnboarding email > renders email template as html string 1`] = `
<div>
  <div>
    <p>
      Bonjour,
    </p>
     
    <p>
      Un compte vous a été créé sur l'application Carnet de bord pour administrer le déploiement 
      <strong>
        déploiement
      </strong>
      .
    </p>
     
    <p>
      Pour accéder à Carnet de bord, veuillez cliquer sur le lien ci-dessous :
    </p>
     
    <p
      style="padding-left: 20%;"
    >
      <a
        href="some-url.com/auth/jwt/access-key?url=redirect-to-url.com"
        rel="nofollow"
        style="background-color: rgb(106, 32, 174); font-size: 14px; font-family: Helvetica, Arial, sans-serif; font-weight: bold; text-decoration: none; padding: 8px 10px; color: rgb(255, 255, 255); border-radius: 5px; display: inline-block;"
      >
        <span>
          Accédez à Carnet de bord
        </span>
      </a>
    </p>
    <!--&lt;AccessButton&gt;-->
     
    <p>
      L'équipe Carnet de bord
    </p>
    <!--&lt;Footer&gt;-->
  </div>
  <!--&lt;ManagerOnboarding&gt;-->
</div>
`;

exports[`creating email > notebookInvitation email > renders email template as html string with creator info 1`] = `
<div>
  <div>
    <p>
      Bonjour 
      Prénom Nom
      ,
    </p>
    <!--&lt;Greeting&gt;-->
     
    <p>
      Vous avez été invité(e) à rejoindre un carnet de bord par
			
      <b>
        Prénom du créateur Nom du créateur
      </b>
      .
    </p>
     
    <p>
      Pour accéder à Carnet de bord, veuillez cliquer sur le lien ci-dessous :
    </p>
     
    <p
      style="padding-left: 20%;"
    >
      <a
        href="some-url.com/auth/jwt/access-key?url=redirect-to-url.com"
        rel="nofollow"
        style="background-color: rgb(106, 32, 174); font-size: 14px; font-family: Helvetica, Arial, sans-serif; font-weight: bold; text-decoration: none; padding: 8px 10px; color: rgb(255, 255, 255); border-radius: 5px; display: inline-block;"
      >
        <span>
          Accédez à Carnet de bord
        </span>
      </a>
    </p>
    <!--&lt;AccessButton&gt;-->
     
    <p>
      L'équipe Carnet de bord
    </p>
    <!--&lt;Footer&gt;-->
  </div>
  <!--&lt;NotebookInvitation&gt;-->
</div>
`;

exports[`creating email > notebookInvitation email > renders email template as html string without creator info 1`] = `
<div>
  <div>
    <p>
      Bonjour 
      Prénom Nom
      ,
    </p>
    <!--&lt;Greeting&gt;-->
     
     
    <p>
      Pour accéder à Carnet de bord, veuillez cliquer sur le lien ci-dessous :
    </p>
     
    <p
      style="padding-left: 20%;"
    >
      <a
        href="some-url.com/auth/jwt/access-key?url=redirect-to-url.com"
        rel="nofollow"
        style="background-color: rgb(106, 32, 174); font-size: 14px; font-family: Helvetica, Arial, sans-serif; font-weight: bold; text-decoration: none; padding: 8px 10px; color: rgb(255, 255, 255); border-radius: 5px; display: inline-block;"
      >
        <span>
          Accédez à Carnet de bord
        </span>
      </a>
    </p>
    <!--&lt;AccessButton&gt;-->
     
    <p>
      L'équipe Carnet de bord
    </p>
    <!--&lt;Footer&gt;-->
  </div>
  <!--&lt;NotebookInvitation&gt;-->
</div>
`;
